### PR TITLE
Clang tidy: modernize use nullptr instead of NULL

### PR DESCRIPTION
**Pull request by clementperon**
This MR add the clang-tidy support + convert NULL to nullptr.

This superseed https://github.com/seladb/PcapPlusPlus/pull/1020

### CODE REVIEW

**Comment by seladb** [comment]
@clementperon the build on centos fails: https://github.com/seladb/PcapPlusPlus/actions/runs/3842342225/jobs/6543558857

And there are a few failures in CirrusCI as well.

In addition - it's hard to review such big PRs, even though the change is pretty straightforward...

**Comment by clementperon** [comment]
@seladb agree that's why I have split into a separate commit and only enabled modernize nullptr. 

This is done automatically by clang-tidy

File: CMakeLists.txt [hunk 1]
```diff
@@ -102,6 +102,14 @@ if(PCAPPP_ENABLE_PCAP_SET_DIRECTION)
   add_definitions(-DHAS_SET_DIRECTION_ENABLED)
 endif()
 
+option(PCAPPP_ENABLE_CLANG-TIDY "Run Clang-Tidy static analysis during build" OFF)
```
Comment (seladb) [inline CMakeLists.txt:105]: nit: I'd change it to `PCAPPP_ENABLE_CLANG_TIDY` (with `-` instead of `_`)

File: CMakeLists.txt [hunk 2]
```diff
@@ -102,6 +102,14 @@ if(PCAPPP_ENABLE_PCAP_SET_DIRECTION)
...
+
+if (PCAPPP_ENABLE_CLANG-TIDY)
+  find_program(CLANG_TIDY_EXE NAMES "clang-tidy" REQUIRED)
+  set(CLANG_TIDY_COMMAND "${CLANG_TIDY_EXE}" "--fix" "--checks=modernize-use-nullptr")
```
Comment (seladb) [inline CMakeLists.txt:109]: - do we only want to check `nullptr`? Maybe enable all checks?
- Do we always want the `--fix` flag or should it be configurable?

Comment (clementperon) [reply CMakeLists.txt:109]: For the moment we only check for nullptr yes, I would like toenable the other checks progressively in another MR

Comment (seladb) [reply CMakeLists.txt:109]: ok sounds reasonable, what about the `--fix` flag?

**Review by seladb — APPROVED** [review]
Just fix this small issue: https://github.com/seladb/PcapPlusPlus/pull/1028#discussion_r1064490751

And merge the PR!

**Pull request merged**